starknet_py/net/client.py

```diff
@@ -155,30 +166,37 @@ async def wait_for_tx(
                 " block have status ACCEPTED_ON_L2."
             )
 
+        transaction_received = False
         while True:
+            retries -= 1
             try:
-                tx_receipt = await self.get_transaction_receipt(tx_hash=tx_hash)
-
-                deprecated_status = _status_to_finality_execution(tx_receipt.status)
-                finality_status = tx_receipt.finality_status or deprecated_status[0]
-                execution_status = tx_receipt.execution_status or deprecated_status[1]
-
-                if execution_status == TransactionExecutionStatus.REJECTED:
-                    raise TransactionRejectedError(message=tx_receipt.rejection_reason)
+                if not transaction_received:
+                    tx_status = await self.get_transaction_status(tx_hash=tx_hash)
+
+                    if tx_status.finality_status == TransactionStatus.REJECTED:
+                        raise TransactionRejectedError()
+
+                    transaction_received = True
+                else:
+                    tx_receipt = await self.get_transaction_receipt(tx_hash=tx_hash)
+
+                    if (
+                        tx_receipt.execution_status
+                        == TransactionExecutionStatus.REVERTED
+                    ):
+                        raise TransactionRevertedError(message=tx_receipt.revert_reason)
+
+                    if (
+                        tx_receipt.execution_status
+                        == TransactionExecutionStatus.SUCCEEDED
+                        or tx_receipt.finality_status
+                        in (
+                            TransactionFinalityStatus.ACCEPTED_ON_L2,
+                            TransactionFinalityStatus.ACCEPTED_ON_L1,
+                        )
+                    ):
+                        return tx_receipt
```

Comment:
`finality_status` is a mandatory property in `tx_receipt` with only two possible values: `ACCEPTED_ON_L2` and `ACCEPTED_ON_L1`. As a result, the given condition will always evaluate to true.

Since you raise the error if the transaction is `REVERTED` above, I don't really see the need for this check:
```suggestion
                    else:
                        return tx_receipt
```
Additionally, you could introduce `isAccepted`/`isSuccessful`/`isReverted` method in `tx_receipt` to encapsulate the condition that makes transaction "unsuccessful".